src/main/java/software/amazon/cloudformation/LambdaWrapper.java

```diff
@@ -487,11 +487,8 @@ private void publishExceptionMetric(final Action action, final Throwable ex, fin
     private void
         publishExceptionCodeAndCountMetric(final Action action, final HandlerErrorCode handlerErrorCode, final boolean thrown) {
         if (this.metricsPublisherProxy != null) {
-            EnumSet.allOf(HandlerErrorCode.class).stream()
-                // publishing 0 value for all (if not thrown) otherwise filtered
-                .filter(errorCode -> errorCode.equals(handlerErrorCode) || !thrown)
-                .forEach(errorCode -> this.metricsPublisherProxy.publishExceptionByErrorCodeMetric(Instant.now(), action,
-                    errorCode, thrown));
+            EnumSet.allOf(HandlerErrorCode.class).forEach(errorCode -> this.metricsPublisherProxy
+                .publishExceptionByErrorCodeMetric(Instant.now(), action, errorCode, thrown && errorCode == handlerErrorCode));
```

Comment:
We should publish all those ErrorCode metrics in one PutMetric call, which can support multiple metrics in one request.. Otherwise it will consume large number of TPS, and may cause throttling.